src/components/MonthCalendar.tsx

```diff
@@ -59,21 +64,47 @@ const MonthCalendar = ({ events, onFocusDate, focusDate }: MonthCalendarProps):
             return 0;
         };
 
-        const lastCalendarDay = new Date(new Date(firstCalendarDay).setDate(firstCalendarDay.getDate() + gridItems));
-
-        const calendarEvents = getEventsInRange(firstCalendarDay, lastCalendarDay);
-
-        if (calendarEvents.length === 0) return { height: 0, top: 0 };
-
-        const firstEventDate = new Date(calendarEvents[0].start_time);
-        const lastEventDate = new Date(calendarEvents[calendarEvents.length - 1].end_time);
-
-        const top = getGridNumber(new Date(firstEventDate)) * barUnit;
-        const height = (getGridNumber(new Date(lastEventDate)) * barUnit) - top;
-
+        const campaignEvents = events.filter(e => e.campaign.id === campId)
+            .sort((a, b) => {
+                return new Date(a.start_time).getTime() - new Date(b.start_time).getTime();
+            });
+
+        if (campaignEvents.length === 0) return { height: 0, top: 0 };
+
+        const firstEventDate = new Date(campaignEvents[0].start_time);
+        const lastEventDate = new Date(campaignEvents[campaignEvents.length - 1].end_time);
+
+        let bottom, top;
+        if (firstEventDate < firstCalendarDay ) {
+            top = 0;
+        }
+        else if (firstEventDate > lastCalendarDay) {
+            top = 100;
+        }
+        else {
+            top = getGridNumber(new Date(firstEventDate)) * barUnit;
+        }
+
+        if (lastEventDate > lastCalendarDay) {
+            bottom = 100;
+        }
+        else if (lastEventDate < firstCalendarDay) {
+            bottom = 0;
+        }
+        else {
+            bottom = getGridNumber(new Date(lastEventDate)) * barUnit;
+        }
+        if (bottom > 100) bottom = 100;
```

Comment:
Avoid these single-line if statements. They're more difficult to read, and we can afford the space.